pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1653,6 +1642,40 @@ func (cc *XController) addQueueJob(obj interface{}) {
 
 	klog.V(6).Infof("[Informer-addQJ] enqueue %s &qj=%p Version=%s Status=%+v", qj.Name, qj, qj.ResourceVersion, qj.Status)
 	cc.enqueue(qj)
+	// Requeue the item to be processed again in 30 seconds.
+	//TODO: tune the frequency of reprocessing an AW
+	hasCompletionStatus := false
+	for _, genericItem := range qj.Spec.AggrResources.GenericItems {
+		if len(genericItem.CompletionStatus) > 0 {
+			hasCompletionStatus = true
+		}
+	}
+	//When an AW entrs a system with completionstatus keep checking the AW until completed
+	//updatequeuejobs now runs as a part of informer machinery. optimization here is to not use etcd to pullout submitted AWs and operate
+	//on stale AWs. This has potential to improve performance at scale.
+	//if qj.Status.State != arbv1.AppWrapperStateCompleted && qj.Status.State != arbv1.AppWrapperStateFailed && qj.Status.State != "" {
+	requeueInterval := 30 * time.Second
+	key, err := cache.MetaNamespaceKeyFunc(qj)
+	if err == nil {
+		go func() {
```

Comment:
`time.AfterFunc` seems a better construct to use here rather than `go func() + time.Sleep`.

On a side node, it seems it'll be valuable to move to used a delaying queue.

Reply:
We need to wake the thread up every so often, from what I understand time.AfterFunc would wake up only once.

Reply:
Right, maybe requeuing within the function passed to `time.AfterFunc` (possibly calling `time.AfterFunc`)?